src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-language.adoc

```diff
@@ -1,27 +1,38 @@
 [id="adding-support-for-a-new-language_{context}"]
 = Adding support for a new language
 
-This paragraph is the procedure module introduction: a short description of the procedure.
+This section describes how to add support for a new language into Eclipse Che.
 
 [discrete]
 == Prerequisites
 
-* A bulleted list of conditions that must be satisfied before the user starts following this assembly.
-* You can also link to other modules or assemblies the user must follow before starting this assembly.
-* Delete the section title and bullets if the assembly has no prerequisites.
+* Install the latest Che-Theia plug-in Yeoman generator by running:
++
+----
+$ npm install -g yo @theia/generator-plugin
+----
 
 [discrete]
 == Procedure
 
-. Start each step with an active verb.
+. Generate a sample by running `yo @theia/plugin`. When prompted, confirm the choices for *Backend plug-in*, *Samples*, and then *Language sample*. This creates a project and a ready-to-use plug-in, which can be deployed.
 
-. Include one command or action per step.
+. Replace the download url of the implementation of language server protocol in `src/download-xml-ls.js` file. This file will be downloaded during the build process of the plug-in. If the name of `src/download-xml-ls.js` was changed, need to replace it in `package.json` in `scripts` block as well.
 
-. Use an unnumbered bullet (*) if the procedure includes only one step.
+. Replace the configuration of the language in `src/language-configuration.json` file, this file should contain configuration options for current language. If the name of `src/language-configuration.json` was changed, need to replace it in `package.json` in `contribution` block as well.
+
+. Replace the configuration of the language in `src/xml.tmLanguage.json` file, this file describes syntax highlighting, determines the color and style of source code displayed in the editor. If the name of `src/xml.tmLanguage.json` was changed, need to replace it in `package.json` in `contribution.grammars` block as well.
+
+. Replace an information about the provided language in `src/plug-in_name.ts`. In that file should be an information about the location of LSP implementation and a command with arguments to run it.
+
+. Build the plug-in by running `yarn` in the root of the project.
+
+. Deploy the Che-Theia plug-in via the plug-in registry.
```

Comment:
I would link to the section about "how to make a che 7 plugin out of a VS Code extension directly here. I didn't catch the connection to the link below, first

Reply:
Added a link to the documentation on how to add a plugin to plugin-registry as a last step